ComplianceAsCode/content
# Make our Ansible content to skip tasks on platforms that are not applicable.

Some rules can define platforms that they are applicable for using the `<platform idref="cpe:/a:platform-id"/>` tag - see #3576.
This PR adds support for `@ANSIBLE_ENSURE_PLATFORM@` macro that expands to a `when: ...` Ansible clause, if the respective rule is bound to a platform.

- If a known platform is encountered, respective `when` entry is created.
- If an unknown platform is encountered, a descriptive error comment is created.
- If no platform is specified, an informative comment is created.

Like this, there won't be any annoying failures when remediating a container with rules that are machine-only.

How to test this:
#. Build SSG.
#. Choose a product and open it's `shorthand.xml`
#. Find a rule that has an Ansible remediation and add e.g. `<platform idref="cpe:/a:machine"/>` as a sub-element of the respective `Rule` element.
#. Re-run `make` - the shorthand won't be regenerated, but changes will be picked up.
#. Check out the HTML guide and expand the Ansible remediation of the concerned rule.

Remarks:
- Only one platform per rule is supported at the moment.
- The 'machine' platform is supported.
- The Ansible detection of whether or not are we running on a container is based on running `ansible -m setup localhost` in a Docker container.
